src/pages/User.tsx

```diff
@@ -0,0 +1,26 @@
+import { Link, useParams } from "react-router-dom";
+import { useUser } from "../hooks/users";
+
+function User(): JSX.Element {
+  const { userId } = useParams();
+
+  const { data, error, isError, isLoading } = useUser(userId || "");
```

Comment:
I would probably use the not-null assertion operator (`!`)

```suggestion
  const { data, error, isError, isLoading } = useUser(userId!);
```

Alternatively you could check for existence and redirect the user if it does not

```tsx
if (!userId) {
  return <Navigate to="/not-found" />
}
```

Reply:
Or better, since this is user-controlled input, you should validate it. You could use `zod` for this as well

```ts
const paramsSchema = z.object({
  userId: z.string(),
})

type Params = z.infer<typeof paramsSchema>

const params = useParams<Params>()
const { userId } = paramsSchema.parse(params)
```